sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -52,9 +53,14 @@ public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
     }
 
     ServiceBusReceivedMessage(BinaryData body) {
+        Objects.requireNonNull(body, "'body' cannot be null.");
         amqpAnnotatedMessage = new AmqpAnnotatedMessage(new AmqpDataBody(Collections.singletonList(body.toBytes())));
     }
 
+    ServiceBusReceivedMessage(byte[] body) {
+        this(BinaryData.fromBytes(Objects.requireNonNull(body, "'body' cannot be null.")));
+    }
```

Comment:
Why do we need this constructor? This instance is always created by the SDK code. So, we can just use BinaryData. I think we should add getBodyAsBytes() for the user to get the byte[] in ServiceBusREceivedMessage.

Reply:
yes this is internal to us and I added it for consistency. But I will remove  this constructor